tests/ModelTest.php

```diff
@@ -657,161 +656,80 @@ public function testDates(): void
-
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('2010-08-08 04.08.37.324')]);
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
+    public static function provideDate(): \Generator
+    {
+        yield [time()];
+        yield [Date::now()];
+        yield ['Monday 8th August 2005 03:12:46 PM'];
+        yield ['Monday 8th August 1960 03:12:46 PM'];
+        yield ['2005-08-08'];
+        yield ['1965-08-08'];
+        yield [new DateTime('2010-08-08')];
+        yield [new DateTime('1965-08-08')];
+        yield [new DateTime('2010-08-08 04.08.37')];
+        yield [new DateTime('1965-08-08 04.08.37')];
+        yield [new DateTime('2010-08-08 04.08.37.324')];
+        yield [new DateTime('1965-08-08 04.08.37.324')];
+        yield [new DateTime('1965-08-08 04.08.37.324')];
+    }
 
-        $user = User::create(['name' => 'Jane Doe', 'birthday' => new DateTime('1965-08-08 04.08.37.324')]);
+    /**
+     * @dataProvider provideDate
+     */
+    public function testDateInputs($date): void
+    {
+        /** @var User $user */
+        // Test with create and standard property
+        $user = User::create(['name' => 'Jane Doe', 'birthday' => $date]);
         $this->assertInstanceOf(Carbon::class, $user->birthday);
 
         //Test with setAttribute and standard property
-        $user->setAttribute('birthday', time());
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', Date::now());
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', 'Monday 8th August 2005 03:12:46 PM');
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', 'Monday 8th August 1960 03:12:46 PM');
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', '2005-08-08');
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', '1965-08-08');
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', new DateTime('2010-08-08'));
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', new DateTime('1965-08-08'));
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', new DateTime('2010-08-08 04.08.37'));
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', new DateTime('1965-08-08 04.08.37'));
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', new DateTime('2010-08-08 04.08.37.324'));
-        $this->assertInstanceOf(Carbon::class, $user->birthday);
-
-        $user->setAttribute('birthday', new DateTime('1965-08-08 04.08.37.324'));
+        $user->setAttribute('birthday', null);
```

Comment:
Is this call necessary? If so, please add a comment to explain why it must remain - this looks like something I'd "optimise away".

Reply:
Yes, to ensure the value modified. Because the value is set to the same previous value. I can add an `assertNull`.

Reply:
Ah, I see. A comment to remind people that the value has to be modified might be useful then, but an assertion would probably do as well. I'll leave it up to you.